rustls/src/server/tls13.rs

```diff
@@ -604,7 +604,8 @@ mod client_hello {
         common.send_msg(m, false);
     }
 
-    #[allow(clippy::needless_pass_by_ref_mut)] // cx only mutated if cfg(feature = "quic")
+    #[allow(unknown_lints)]
```

Comment:
The problem with CI commands that I don't like having to remember the complete incantation when I'm running clippy locally. I'd prefer turning this into a `#![allow(unknown_lints]` in the crate root, similar to what we do for other lint tuning.

Reply:
That makes sense. Running `cargo clippy` locally the warnings aren't denied but default but it is noise that could confuse someone new to the project, so I think you've convinced me it's a good idea to fix at the site of the issue.

Reply:
I think it's there because it allows the unknown lints on stable, but not on nightly, which is a bit more robust in the sense that we'll notice when a lint is actually unknown. But on the other hand, if a lint is ever renamed on nightly so it doesn't exist there anymore but still exists on stable, it will also break... So yes, I think the local solution is probably better, because at least that way we get to avoid that we don't notice unknown lints at all.